### PR TITLE
Create answerfile class, clean Cluster::fillData and improve answerfile code

### PR DESCRIPTION
**Pull request by lbgracioso**
Hey! This is a HUGE one.

Created a new class to add verification to the answerfile before using it and also to clean Cluster::fillData (it was a whole mess).

If you guys have any verifications in mind that I can add, just let me know!

### CODE REVIEW

File: include/cloysterhpc/inifile.h [hunk 1]
```diff
@@ -17,7 +17,8 @@ class inifile {
     void loadFile(const std::string& filepath);
     void loadFile(std::string_view filepath);
     void loadFile(const std::filesystem::path& filepath);
-    std::string getValue(const std::string& section, const std::string& key);
+    std::string getValue(const std::string& section, const std::string& key,
+        const bool& optional = true);
```
Comment (viniciusferrao) [inline include/cloysterhpc/inifile.h:21]: `bool` is a primitive type. Don't need to const ref it.

File: src/answerfile.cpp [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+            throw std::runtime_error(
+                fmt::format("Section 'network_management' field 'gateway' "
+                            "validation failed - {}",
+                    e.what()));
+        }
+    }
+
+    if (m_ini.exists("network_management", "nameservers")) {
+        std::vector<std::string> nameservers;
+        boost::split(nameservers,
```
Comment (viniciusferrao) [inline src/answerfile.cpp:141]: Have you tried using `<ranges>` from C++20?
https://en.cppreference.com/w/cpp/ranges/split_view

File: src/answerfile.cpp [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+
+AnswerFile::AFNode AnswerFile::validateNode(AnswerFile::AFNode node)
+{
+    if (node.start_ip->is_unspecified()) {
+        if (nodes.generic->start_ip->is_unspecified()) {
+            throw std::runtime_error(
+                fmt::format("Node must have a \"{0}\" key or you must inform a "
+                            "generic \"{0}\" value",
+                    "node_start_ip"));
+        } else {
```
Comment (viniciusferrao) [inline src/answerfile.cpp:322]: This `else` is unnecessary. Since you're throwing inside the `if` block.

File: src/inifile.cpp [hunk 4]
```diff
@@ -22,9 +23,14 @@ void inifile::loadFile(const std::filesystem::path& filepath)
 }
 
 std::string inifile::getValue(
-    const std::string& section, const std::string& key)
+    const std::string& section, const std::string& key, const bool& optional)
```
Comment (viniciusferrao) [inline src/inifile.cpp:26]: Just pass by copy.

**Review by viniciusferrao — COMMENTED** [review]
Hi @lbgracioso can you take a look on my comments? Also do you mind in adding some `doxygen` style documentation on the methods? I'm a bit lost about the usability of some of them.

File: src/answerfile.cpp (hunk 2 above)
Comment (lbgracioso) [reply src/answerfile.cpp:141]: No... Though I don't see how it would help.

**Comment by lbgracioso** [comment]
@viniciusferrao Please check https://github.com/viniciusferrao/cloysterhpc/pull/25/commits/b1da6d773ded4e35340a4e8bdf36c41a5640ba16

About the `doxygen` documentation, can you tell me which ones you think should have? I don't see the needing since most of them are "load" methods, loading data from the answerfile..

**Comment by viniciusferrao** [comment]
> @viniciusferrao Please check [b1da6d7](https://github.com/viniciusferrao/cloysterhpc/commit/b1da6d773ded4e35340a4e8bdf36c41a5640ba16)
> 
> About the `doxygen` documentation, can you tell me which ones you think should have? I don't see the needing since most of them are "load" methods, loading data from the answerfile..

LGTM.

**Pull request merged**